lib/1.4/dml-builtins.dml

```diff
@@ -519,8 +544,10 @@ template device {
     // this carried semantics in DML 1.2; deprecated in 1.4
     param _confidentiality          = undefined;
 
+    independent method _model_init() { _rec_model_init(); }
     method _init() { _rec_init(); }
     method _post_init() { _rec_post_init(); }
+    independent method model_init() default {}
     method init() default {}
     method post_init() default {}
```

Comment:
these should be `shared`?

Reply:
i dunno, I'm not the one who wrote it. It really doesn't matter, it's not like the device template will be instantiated by more than one object.

Reply:
There is also a slight semantic difference: `template x is device { shared independent method model_init() {} }` is permitted iff the method has the `shared` annotation. I agree that this does not make a difference in practice.